src/js/components/DateInput/DateInput.js

```diff
@@ -262,19 +304,23 @@ Use the icon prop instead.`,
         <Keyboard
           onEsc={open ? () => closeCalendar() : undefined}
           onSpace={(event) => {
-            event.preventDefault();
-            openCalendar();
+            if (!readOnlyCopy) {
+              event.preventDefault();
+              if (!readOnly) openCalendar();
+            }
           }}
         >
-          <Box
+          <StyledDateInputContainer
             ref={containerRef}
             border={!plain}
             round={theme.dateInput.container.round}
             direction="row"
+            readOnlyProp={readOnly}
```

Comment:
Are we calling this `readOnlyProp` to avoid it being passed through to DOM? If so, maybe a comment just so it doesn't get changed in future

Reply:
Yes, I'll add a comment

Reply:
done